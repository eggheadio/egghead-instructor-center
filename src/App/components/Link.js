import React from 'react'

const Link = ({
  url,
  children,
}) => (
  <a
    href={url}
    target='_blank'
    className='blue'
  >
    {children}
  </a>
)

Link.propTypes = {
  url: React.PropTypes.string.isRequired,
  children: React.PropTypes.node,
}

export default Link
