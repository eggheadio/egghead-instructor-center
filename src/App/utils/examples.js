export default {
  text: {
    short: 'Lorem ipsum',
    medium: 'Lorem ipsum dolor sit amet, urna erat feugiat egestas, metus molestie, adipiscing sed nisl et, dictumst erat sem erat.',
  },
  url: 'https://egghead.io',
  markdown: 'Lorem ipsum *dolor*',
}
