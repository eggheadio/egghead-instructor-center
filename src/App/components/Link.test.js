import React from 'react'
import renderer from 'react-test-renderer'

import examples from '../../App/utils/examples'

import Link from './Link'

test('Default', () => {
  const tree = renderer.create(
    <Link url={examples.url}>
      {examples.text.short}
    </Link>
  )
  expect(tree).toMatchSnapshot()
})
