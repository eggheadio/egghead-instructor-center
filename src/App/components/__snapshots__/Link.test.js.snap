exports[`test Default 1`] = `
<a
  className="blue"
  href="https://egghead.io"
  target="_blank">
  Lorem ipsum
</a>
`;
